### PR TITLE
Interface to create a new flight

### PR DESCRIPTION
**Pull request by davidfischer**
This interface basically just lets you set the name and then sends you to the more complex update screen.

### CODE REVIEW

File: adserver/forms.py [hunk 1]
```diff
@@ -313,6 +314,64 @@ class Meta:
...
+
+    def generate_slug(self):
+        campaign_slug = self.instance.campaign.slug
+        slug = slugify(self.instance.name)
+        if not slug.startswith(campaign_slug):
+            slug = slugify(f"{campaign_slug}-{slug}")
+
+        while Flight.objects.filter(slug=slug).exists():
+            random_char = get_random_string(1)
+            slug = slugify(f"{slug}{random_char}")
```
Comment (ericholscher) [inline adserver/forms.py:357]:
```suggestion
            slug = slugify(f"{slug}-{random_char}")
```

To make it a bit cleaner?

File: adserver/forms.py [hunk 2]
```diff
@@ -313,6 +314,64 @@ class Meta:
...
+            Fieldset(
+                "",
+                "name",
+                "campaign",
+                css_class="my-3",
+            ),
+            Submit("submit", _("Create flight")),
+            HTML(
+                "<p class='form-text small text-muted'>"
+                + str(_("You will be able to edit details on the next screen"))
```
Comment (ericholscher) [inline adserver/forms.py:344]: 👍

File: adserver/templates/adserver/advertiser/flight-list.html [hunk 3]
```diff
@@ -18,6 +18,16 @@
 <h1>{% block heading %}{% trans "Manage advertising flights" %}{% endblock heading %}</h1>
 
 
+<aside class="col text-right">
+  {% if request.user.is_staff %}
+    <a href="{% url 'flight_create' advertiser.slug %}" class="btn btn-sm btn-outline-secondary mb-3" role="button" aria-pressed="true">
+      <span class="fa fa-plus mr-1" aria-hidden="true"></span>
+      <span>{% trans 'Create flight' %}</span>
```
Comment (ericholscher) [inline adserver/templates/adserver/advertiser/flight-list.html:25]: Should the UI make it more obvious this is an admin-only feature currently? Maybe a specific color or lock icon or something?

File: adserver/templates/adserver/advertiser/flight-list.html [hunk 4]
```diff
@@ -18,6 +18,16 @@
 <h1>{% block heading %}{% trans "Manage advertising flights" %}{% endblock heading %}</h1>
 
 
+<aside class="col text-right">
+  {% if request.user.is_staff %}
```
Comment (ericholscher) [inline adserver/templates/adserver/advertiser/flight-list.html:22]: Should this use the video recording permission, so users don't get confused and want it?

**Review by ericholscher — APPROVED** [review]
Simple enough. I do think we should probably do a bit more to differentiate staff actions in the UI, since we've been confused here before (eg. on recording videos -- we should probably use a similar permission so users don't ask questions).

File: adserver/templates/adserver/advertiser/flight-list.html (hunk 4 above)
Comment (davidfischer) [reply adserver/templates/adserver/advertiser/flight-list.html:22]: Probably just check "flight create" permissions more precisely.

File: adserver/templates/adserver/advertiser/flight-list.html (hunk 3 above)
Comment (davidfischer) [reply adserver/templates/adserver/advertiser/flight-list.html:25]: Longer term, I do think we'll be opening these up to users but maybe an indication isn't bad. I'll see what I can do.

Comment (davidfischer) [reply adserver/templates/adserver/advertiser/flight-list.html:25]: I added a few indicators -- both color and icon.

**Pull request merged**